### PR TITLE
Feature : create a new component to display  status of submissions

### PR DESCRIPTION
**Pull request by SirineMhedhbi**
## Context

See : https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/340

## Changes

- [x] Add icons, for the  statuses: 
- Parsed successfully = > (Parsed, Indexed, Metrics, Annotator, Diff) or (Parsed, Indexed, Metrics, Annotator) => green 
- Error in parsing => error parsing => Red
- Archived => Gray
- Others status => sentence with error ( for example : (Parsed, Indexed, Metrics, Annotator, Error Diff)) => Orange

image

image

image

image

- [x] Create a component to display status of submissions
- [x]  Use this component in the submission table
- [x]  Use the component to display icons in front of the name of each ontology on the browse page (only for admin ).

image

- Logged (admin) :
image

- Not logged:
image

### CODE REVIEW

File: app/assets/images/icons/error.svg
Comment (syphax-bouazzouni) [inline app/assets/images/icons/error.svg:1]: This icon was used in other views, you need to be sure that this change does not break the other views/components

File: app/assets/images/icons/success.svg
Comment (syphax-bouazzouni) [inline app/assets/images/icons/success.svg:1]: This icon was used in other views, you need to be sure that this change does not break the other views/components

File: app/helpers/ontologies_helper.rb [hunk 1]
```diff
@@ -148,16 +148,18 @@ def status_link(submission, latest = false, target = '')
     version_link + status_text
   end
 
-  def submission_status2string(sub)
-    return '' if sub.submissionStatus.nil?
+
+  def submission_status2string(data)
```
Comment (syphax-bouazzouni) [inline app/helpers/ontologies_helper.rb:152]: this function was used in other views, you need to be sure that this change does not break it

File: app/views/submissions/_submissions.html.haml [hunk 2]
```diff
@@ -39,7 +39,8 @@
 
           - if @ontology.admin?(session[:user])
             - r.td { raw sub.submissionId }
-          - r.td { raw status_link(sub, sub.submissionId==submission_readyId)}
+          - r.td do
```
Comment (syphax-bouazzouni) [inline app/views/submissions/_submissions.html.haml:42]: you don't need to add a column for the status, why not just append it to the version column?

File: app/assets/images/icons/alert-triangle.svg
Comment (syphax-bouazzouni) [inline app/assets/images/icons/alert-triangle.svg:1]: I think this icon need to be in orange.

File: app/assets/images/icons/archive.svg
Comment (syphax-bouazzouni) [inline app/assets/images/icons/archive.svg:1]: I think this icon needs be in light gray

File: app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml [hunk 3]
```diff
@@ -4,6 +4,8 @@
       .browse-ontology-title-bar
         %a.browse-ontology-title{:href => "/ontologies/#{ontology[:acronym]}", data: {'turbo-frame': '_top'}}
           = ontology[:name]+" ("+ontology[:acronym]+")"
+          - ontology_status = status_string(ontology)
```
Comment (syphax-bouazzouni) [inline app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:7]: show this only if you logged in and are admin

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/helpers/ontologies_helper.rb (hunk 1 above)
Comment (SirineMhedhbi) [reply app/helpers/ontologies_helper.rb:152]: I've double-checked the changes, and they're working smoothly across all views without any problems

File: app/assets/images/icons/alert-triangle.svg
Comment (SirineMhedhbi) [reply app/assets/images/icons/alert-triangle.svg:1]: i will change it

File: app/assets/images/icons/archive.svg
Comment (SirineMhedhbi) [reply app/assets/images/icons/archive.svg:1]: i will change it

File: app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml (hunk 3 above)
Comment (SirineMhedhbi) [reply app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:7]: okey

File: app/views/submissions/_submissions.html.haml (hunk 2 above)
Comment (SirineMhedhbi) [reply app/views/submissions/_submissions.html.haml:42]: I tried that approach, but it caused an issue

File: app/assets/images/icons/error.svg
Comment (SirineMhedhbi) [reply app/assets/images/icons/error.svg:1]: i will change the icon

File: app/assets/images/icons/success.svg
Comment (SirineMhedhbi) [reply app/assets/images/icons/success.svg:1]: i will change the icon

File: app/views/submissions/_submissions.html.haml (hunk 2 above)
Comment (syphax-bouazzouni) [reply app/views/submissions/_submissions.html.haml:42]: OK, it's not important.

File: app/assets/images/icons/success.svg
Comment (syphax-bouazzouni) [reply app/assets/images/icons/success.svg:1]: You can use the same Icon, but just change the width, in your code not in the original code.
Something like this 
```ruby
= inline_svg('icons/success', width: '21', height: '21)
```

Comment (SirineMhedhbi) [reply app/assets/images/icons/success.svg:1]: okey , thanks

File: app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml (hunk 3 above)
Comment (syphax-bouazzouni) [reply app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:7]: Can be simplified to this `session[:user]&.admin?`

Comment (SirineMhedhbi) [reply app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:7]: okey

**Comment by jonquet** [comment]
Sounds very good to me. I like the icons and integration. 
Just a note: @SirineMhedhbi do we have the old text with details on the status on mouse overing the icon?

**Comment by SirineMhedhbi** [comment]
@jonquet i didn't change the text 
here are all the possible values ​​of submit status => https://github.com/ontoportal-lirmm/ontologies_linked_data/blob/master/lib/ontologies_linked_data/models/submission_status.rb#L4

**Review by syphax-bouazzouni — APPROVED** [review]

File: app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml [hunk 4]
```diff
@@ -4,8 +4,8 @@
       .browse-ontology-title-bar
         %a.browse-ontology-title{:href => "/ontologies/#{ontology[:acronym]}", data: {'turbo-frame': '_top'}}
           = ontology[:name]+" ("+ontology[:acronym]+")"
-          - if ontology[:private]
-            %i.fas.fa-key{title:"Private Ontology"}
+          - ontology_status = status_string(ontology)
+          = render Display::InfoTooltipComponent.new(text: ontology_status, icon: submission_status_icons(ontology_status)) if session[:user]&.admin?
```
Comment (syphax-bouazzouni) [inline app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:8]: you removed the private ontology icon here, put it again.

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml (hunk 4 above)
Comment (SirineMhedhbi) [reply app/components/ontology_browse_card_component/ontology_browse_card_component.html.haml:10]: Done, thank you for the comment ^^

**Review by syphax-bouazzouni — APPROVED** [review]
Approved, but will not be merged to development until the next release.

**Comment by jonquet** [comment]
Good to me too. Next round of feedback when on StagePortal.

**Comment by syphax-bouazzouni** [comment]
@SirineMhedhbi you can deploy it in Stageportal.

**Comment by SirineMhedhbi** [comment]
@syphax-bouazzouni  okay

**Pull request merged**